tools/utils/utils.js

```diff
@@ -457,9 +457,16 @@ exports.isUrlWithFileScheme = function (x) {
 };
 
 exports.isUrlWithSha = function (x) {
-  // For now, just support http/https, which is at least less restrictive than
-  // the old "github only" rule.
+  // Is a URL with a fixed SHA? We use this for Cordova -- although theoretically we could use
+  // a URL like isNpmUrl(), there are a variety of problems with this, 
+  // see https://github.com/meteor/meteor/pull/5562
   return /^https?:\/\/.*[0-9a-f]{40}/.test(x);
+}
+
+exports.isNpmUrl = function (x) {
+  // These are the various protocols that NPM supports, which we use to download NPM dependencies
+  // See https://docs.npmjs.com/files/package.json#git-urls-as-dependencies
+  return exports.isUrlWithSha(x) || /^(git|git\+ssh|git\+http|git\+https)?:\/\/.*#/.test(x);
```

Comment:
Ah, I didn't get the difference between `https://` and `git+https://`. With Cordova, you use `https://` to download from Git, so that actually supports `#commit-ish` URLs.

(So it allows `https://github.com/apache/cordova-plugin-console.git#f055daec45575bf08538f885e09c85a0eba363ff` or `https://github.com/apache/cordova-plugin-console.git#r0.2.0`).